src/middlewared/middlewared/alembic/versions/12.1/2021-03-15_14-22_remove_rawdisks_from_iscsi.py

```diff
@@ -0,0 +1,48 @@
+"""remove raw disks from iscsi
+
+Revision ID: 370ff38939fd
+Revises: 382b7ca9bb51
+Create Date: 2021-03-15 14:22:19.009163+00:00
+
+"""
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = '370ff38939fd'
+down_revision = '382b7ca9bb51'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    conn = op.get_bind()
+    main_table = 'services_iscsitargetextent'
+    main_col = 'iscsi_target_extent_type'
+    tartoext_table = 'services_iscsitargettoextent'
+
+    tar_ids = set()
+    # query db for entries that point to raw disks
+    for disk_id, in conn.execute(f'SELECT id FROM {main_table} WHERE {main_col} == "Disk"').fetchall():
+        # delete the target to extent associations first
+        for id, tar_id in conn.execute(
+            f'SELECT id, iscsi_target_id FROM {tartoext_table} WHERE iscsi_extent_id == {disk_id}'
+        ).fetchall():
+            tar_ids.add(tar_id)
+            conn.execute(f'DELETE FROM {tartoext_table} WHERE id == {id}')
```

Comment:
Hmm I never saw SQL queries with `==`

Reply:
> Hmm I never saw SQL queries with `==`

sqlite> SELECT * FROM services_iscsitargetextent where id == 2;
                                 id = 2
           iscsi_target_extent_name = disk01
         iscsi_target_extent_serial = 5624ba25976f000
           iscsi_target_extent_type = Disk
           iscsi_target_extent_path = {serial}drive-scsi1
       iscsi_target_extent_filesize = 0
      iscsi_target_extent_blocksize = 512
     iscsi_target_extent_pblocksize = 0
iscsi_target_extent_avail_threshold =
        iscsi_target_extent_comment = disk01
            iscsi_target_extent_naa = 0x6589cfc000000857b205f610a35fb3bd
   iscsi_target_extent_insecure_tpc = 1
            iscsi_target_extent_xen = 0
            iscsi_target_extent_rpm = SSD
             iscsi_target_extent_ro = 0
        iscsi_target_extent_enabled = 1
         iscsi_target_extent_vendor = TrueNAS

Reply:
I did not know that works either. We don't use `==` in SQL queries

Reply:
So are you all asking me to change to `=`?

Reply:
Yes, `==` is not SQL standard as far as I can understand, maybe only works in sqlite3, its not consistent

```
postgres=# select 1=1;
 ?column? 
----------
 t
(1 row)

postgres=# select 1==1;
ERROR:  operator does not exist: integer == integer
LINE 1: select 1==1;
                ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
```